.github/workflows/lint.yml

```
@@ -0,0 +1,75 @@
+          go-version: '1.24.0'
+
+      - name: Install jq
+        run: sudo apt-get update && sudo apt-get install -y jq
+
+      - name: Fetch OpenAPI spec and generate provider (for linting)
+        run: |
+          ./scripts/fetch-openapi-spec.sh
+          ./scripts/generate-provider.sh
+
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
```

actionlint (1.7.7)

the runner of "golangci/golangci-lint-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)